packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -38,6 +38,32 @@ AssertExists() {
   return 0
 }
 
+ParseFlutterBuildMode() {
+  # Use FLUTTER_BUILD_MODE if it's set, otherwise use the Xcode build configuration name
```

Comment:
oh cool, is this a whole new feature (that confused matt forever) bundled here? :D

Reply:
This has been in there for .. a while now.

Basically, we try to use the $CONFIGURATION because that's probably what the user wants if they're running from Xcode (in particular, we don't want instruments tests from debug mode Dart, and we set up instruments to run in a "PROFILE" configuration by default).

But we do let a user override it if necessary.

Reply:
Yeah this logic and error messaging was just moved into a common function so I could use it in two places.

Reply:
we should go have Matt write this down in https://flutter.dev/docs/deployment/flavors :D